beacon-chain/core/blocks/payload.go

```diff
@@ -257,7 +283,11 @@ func ValidatePayloadHeader(st state.BeaconState, header *enginev1.ExecutionPaylo
 }
 
 // ProcessPayloadHeader processes the payload header.
-func ProcessPayloadHeader(st state.BeaconState, header *enginev1.ExecutionPayloadHeader) (state.BeaconState, error) {
+func ProcessPayloadHeader(st state.BeaconState, data interfaces.ExecutionData) (state.BeaconState, error) {
+	header, ok := data.Proto().(*enginev1.ExecutionPayloadHeader)
```

Comment:
I'm not sure if this check is future-proof. What if we have `enginev2.ExecutionPayloadHeader`. Would we just keep adding it?

Reply:
Agreed. Unfortunately, this needs the concrete enginev1.ExecutionPayloadHeader because of the beacon state setter function a few lines below. Perhaps the type assertion can be moved down to the state setter implementation